src/server/sample_data/routes/install.js

```diff
@@ -0,0 +1,103 @@
+import Joi from 'joi';
+
+import { loadData } from './lib/load_data';
+import { createIndexName } from './lib/create_index_name';
+import { adjustTimestamp } from './lib/adjust_timestamp';
+
+export const createInstallRoute = () => ({
+  path: '/api/sample_data/{id}',
+  method: 'POST',
+  config: {
+    validate: {
+      params: Joi.object().keys({
+        id: Joi.string().required(),
+      }).required()
+    },
+    handler: async (request, reply) => {
+      const server = request.server;
+      const sampleDataset = server.getSampleDatasets().find(sampleDataset => {
+        return sampleDataset.id === request.params.id;
+      });
+      if (!sampleDataset) {
+        return reply().code(404);
+      }
+
+      const { callWithRequest } = server.plugins.elasticsearch.getCluster('data');
+      const index = createIndexName(server, sampleDataset.id);
+      const insertCmd = {
+        index: {
+          _index: index
+        }
+      };
+
+      try {
+        await callWithRequest(request, 'indices.delete', { index: index });
```

Comment:
The index name will always be prefixed by `kibana_sample_data_` so at worst, the delete would remove all of the sample data indices but never any real data indices. 

```
export function createIndexName(server, sampleDataSetId) {
  return `kibana_sample_data_${sampleDataSetId}`;
}
```

My thinking was that I wanted the API to be as easy for the user as possible. If an the index already exists, then why make the user send a delete request and then another install request? Their intent by sending the install request was to install the index. Why annoy users with errors that do not matter with the stated API call? The install updates the timestamps to the present so I could see someone wanted to install an already installed index for this purpose.

Reply:
> The index name will always be prefixed by kibana_sample_data_ so at worst, the delete would remove all of the sample data indices but never any real data indices.

Ah yes, I missed that the first time around!  I suppose if we make the prefix configurable it could still delete all data but now we are adding more steps that the user would have to explicitly take to mess it up. 

I still don't know if I'm convinced it's helpful enough to leave in.  A CRUD API has create and update separately.  For the `bulkCreate` Api from saved objects, you have to explicitly pass `overwrite: true`.  I don't think create should have a destructive side affect unless explicitly asked for (e.g. `overwrite: true` flag).  I don't think it's necessary or helpful enough to warrant the risk. The UI requires uninstalling before re-installing - why not keep the API the same?  Unless you think that updating/re-installing the sample data would be common enough to expose in the UI too?

What could potentially happen also is that user A has read/write to sample data and kibana, installs the sample data.  User B has read/write access to Kibana but only read access to the sample data set.  They run this API and the saved objects are updated to new timestamps but the data is old, so it wouldn't be updated appropriately.  But that could be an issue anyway if they called the uninstall API directly so perhaps not important to this decision.

Another point to get rid of it, is that we aren't taking advantage of the functionality ourselves, because the UI requires uninstalling first.  Means the only way to check if it works as expected is to add tests, otherwise no one using Kibana will run into bugs in that path (e.g. what if we handle the permissions case above in the uninstall endpoint but not here - we wouldn't know until someone using the API directly ran into it and filed a bug).

Lets pull in a third party and get another opinion?  I could certainly be overthinking it!  @chrisdavies - do you have any opinion on this matter to help us come to a consensus?

Reply:
I will remove the delete step. I used it a lot when first developing the API when the GUI did not exist. Now that the GUI is there, I doubt many users will actually use the API any ways.